child_compassion/models/project_compassion.py

```diff
@@ -733,6 +734,61 @@ def reactivate_project(self):
             )
         return True
 
+    def sync_projects_from_gmc(
+        self, requests_throttle_seconds=1.0, log_period=100, max_projects_to_sync=-1
+    ):
+        """
+        Synchronises the informations and lifecycle events fro all the projects with
+        active sponsorships from the GMC. This should be called from a cron job and can
+        take a long time to execute (a few hours). The reason for this is that the
+        requests to the GMC server are delayed in order to avoid overwhelming their
+        infrastructure.
+
+        Args:
+            requests_throttle_seconds (float, optional): Time to wait between requests,
+                in seconds. This prevents the GMC server from being overwhelmed.
+                Defaults to 1.0.
+            log_period (int, optional): Write to the log every log_period requests.
+                Defaults to 100.
+            max_projects_to_sync (int, optional): Maximum number of projects to sync.
+                Used for testing. Defaults to -1, which means: sync everything.
+        """
+        # log every so many updated projects to the console
+        projects = self.search([])
+        projects_to_sync = list(filter(lambda p: p.sponsorships_count > 0, projects))
+
+        if max_projects_to_sync == 0:
+            raise ValueError(
+                f"Invalid parameter value {max_projects_to_sync=}. "
+                "Should be either -1 or strictly positive."
+            )
+        max_projects_to_sync = min(max_projects_to_sync, len(projects_to_sync))
+        if max_projects_to_sync > 0:
+            projects_to_sync = projects_to_sync[:max_projects_to_sync]
+
+        nb_projects_to_sync = len(projects_to_sync)
```

Comment:
I would suggest a few changes in order to reduce a bit the number of projects taken into account. Why not focusing on those that were not updated since one month? I would also change the recurring interval of the CRON to one month as well since it's a very costly method.
```suggestion
        one_month_ago = fields.Date.today() - relativedelta(days=30)
        projects_to_sync = self.search([
            ("status", "in", ["Active", "Suspended"]),
            ("last_update_date", "<", one_month_ago)
            ("sponsorships_count", ">", 0)
        ], limit=max_projects_to_sync)
        nb_projects_to_sync = len(projects_to_sync)
```

For this to work you also need to move all these additions in the module `sponsorship_compassion` instead of `child_compassion`, because the field `sponsorships_count` is only available there. You also need to make the field searchable by adding the `search` attribute or by storing it database (I would prefer the first option).

In that case also the parameter `max_projects_to_sync` is no longer tested but passed directly in the search method. An error will be raised from there if it's invalid.